www/js/survey/enketo-survey-service.js

```diff
@@ -174,6 +188,7 @@ angular.module('emission.enketo-survey.service', [
     displayForm: displayForm,
     validateForm: validateForm,
     getAllSurveyAnswers: getAllSurveyAnswers,
+    getAllLocalSurveyAnswers: getAllLocalSurveyAnswers,
```

Comment:
Not sure why you need `getAllLocalSurveyAnswers`. `getAllSurveyAnswers` uses the UnifiedDataLoader, which loads data from both local and remote.

Reply:
I guess he doesn't want the loading message to come up every time a trip end survey has been filled?

Reply:
but why does he need to retrieve anything in the first place? when the survey is complete, the javascript gets back the survey response object, which it saves as a `manual/confirm_survey` object. Why can't you just use that object directly?

I thought that is what was happening before. Thanks for the image of the fix, but I don't understand the problem to begin with 😦

Reply:
There was a report that the button label on the trip does not refresh after the survey is completed.
Normally we reference to the answer with `tripgj.userSurveyanswer`. But this time the rendering of the button is fairly complex so we introduce `tripgj.customSurveyLabel` (see: https://github.com/asiripanich/e-mission-phone/blob/6394a824201bd25e3abc6476e4bd8fee4060d50d/www/js/diary/list.js#L220-L258).

After we move to `tripgj.customSurveyLabel`, the button label stop to automatically refresh because we need to update it manually. So we have to find the right time to do that.

I recalled that I have implemented `EnketoSurvey.launch` to return a promise. So I decided to put the refresh code in there.

I audited the `EnketoSurvey.launch` and it looks like it does not return the answer. It simply return the modal state which is not usable for us.

We wanted to get the latest answer, we have `EnketoSurvey.getAllSurveyAnswers`. But this function makes a call to the server which is fairly expensive.

So I implement `EnketoSurvey.getAllLocalSurveyAnswers` to aid this concern.

Reply:
> After we move to tripgj.customSurveyLabel, the button label stop to automatically refresh because we need to update the variable manually. So we have to find the right time to do that.

I don't buy this. Both `userSurveyAnswer` and `customSurveyLabel` are variables. `customSurveyLabel` is completely dependent on `userSurveyAnswer`. In the old `$scope.populateSurveyAnswerFromTimeline`, you set `tripgj.userSurveyAnswer`, now you are setting `tripgj.customSurveyLabel`. There is no reason why `tripgj.customSurveyLabel` should behave differently.

Although `EnketoSurvey.getAllLocalSurveyAnswers` is a local call, it is still a database call + refresh. Even with local data, too many calls can seriously affect performance as we saw in
https://github.com/e-mission/e-mission-phone/pull/553#issuecomment-482313867

Reply:
In the javascript, `tripgj.userSurveyAnswer` is only accessed in `populateSurveyAnswerFromTimeline`. I really don't see how translating `tripgj.userSurveyAnswer` -> `tripgj.customSurveyLabel` should cause anything to break. 

At worst, you can just choose to not introduce `tripgj.customSurveyLabel` and get it to work that way.

Reply:
I didn't buy this at first as well. But some how `tripgj.customSurveyLabel` break the UI update. That's why I implement this fix.

Reply:
@shankari if you see the earlier PR of this branch, the trip label didn't work properly as @atton16 already explained in this PR. If you don't agree with his approach, could you give a direction for us to fix this issue that doesn't a huge wait time like changing between dates?

Reply:
Apparently, @atton16 can't figure out why `customSurveyLabel` doesn't work after we updated to the new survey so he implemented a fix, which he already tested.

Reply:
> that doesn't a huge wait time like changing between dates?

I am trying to avoid even the call to the local database, so definitely not proposing a huge wait time

Reply:
> I didn't buy this at first as well. But some how tripgj.customSurveyLabel break the UI update. That's why I implement this fix.

As a naive approach, can you go back to filling out `tripgj.userSurveyAnswer`? Yes, the logic to do so might be more complex, but it won't require a database call. If you want to keep the code clean, you can create a local variable (`userSurveyAnswerOld`) that you set at the beginning of the function and use to populate `tripgj.userSurveyAnswer`.

Reply:
Alternatively, put breakpoints in the code, step through, and see how `tripgj.customSurveyLabel` is modified.

Reply:
basically, I think you need to convert "X doesn't work" to "X doesn't work because of Y and there is no workaround for Y" before you decide to go to the database.